kano_world/session.py

```diff
@@ -331,21 +346,24 @@ def upload_share(self, file_path, title, app_name, featured):
                 'attachment': open(file_path, 'rb'),
             }
 
-            # List of attachments to search for in (name, extension) format
+            # TODO A config file would be better suited for such a structure
+            # List of attachments to search for in (name, [extensions...]) fmt
             attachment_files = [
-                ('cover', 'png'),
-                ('resource', 'tar.gz'),
-                ('sample', 'mp3')
+                ('cover', ['png', 'gif']),
+                ('resource', ['tar.gz']),
+                ('sample', ['mp3', 'ogg'])
             ]
 
             for attachment in attachment_files:
-                key, ext = attachment
-                attachment_path = "{}.{}".format(extensionless_path, ext)
+                key, extensions = attachment
+                for ext in extensions:
+                    attachment_path = "{}.{}".format(extensionless_path, ext)
 
-                if os.path.exists(attachment_path):
-                    logger.debug(
-                        'uploading {}: {}'.format(key, attachment_path))
-                    files[key] = open(attachment_path, 'rb')
+                    if os.path.exists(attachment_path):
+                        logger.debug(
+                            'uploading {}: {}'.format(key, attachment_path))
+                        files[key] = open(attachment_path, 'rb')
+                        break
```

Comment:
Hmm, would there ever be the case where you might have more than one extension present there? If so, you might not know which one gets uploaded.